### DISCUSSION TITLE
Is it possible that hist combine same ustack from different process?

### DISCUSSION THREAD

**Discussion by haihuayang** [bpftrace/bpftrace#2488] [Q&A]
Hi, 
I am using bpftrace to report hist like 
    @held_time_ns[arg0, ustack(12)] = hist(nsecs - @held_start[tid, arg0]);
it report multiple entries with the same argument and stack track, I guess that it is because the entries are from different processes. is it possible to make it report once for each combination of argument and stack track?
Regards,

**Comment by viktormalik** [comment]
I believe that this is a bug - there should be a single map entry for each combination of `arg0` and `ustack(12)`. 

It looks like we're not handling the limit correctly. Running:
```
# bpftrace -e 'kfunc:vfs_open { @[ustack(1)] = count(); }
Attaching 1 probe...
^C

@[
    __GI___libc_open+88
]: 1
@[
    __GI___libc_open+88
]: 1
@[
    __GI___libc_open+88
]: 1
[...]
```
Could you verify that this is indeed the problem? I.e. run without the limit and you should see no duplicate keys. If this is the case, please open an issue with some minimal reproducer.

Thanks!

**Comment by haihuayang** [comment]
Hi viktormalik,
I do not know what is the limit you mentioned, could you share the instruction to run without the limit?
Thanks,

**Reply by viktormalik** [reply]
I meant using `ustack` instead of `ustack(12)`. The `12` is the stack depth limit and apparently we're not handling it well.
So, if you use:
```
@held_time_ns[arg0, ustack] = hist(nsecs - @held_start[tid, arg0]);
```
in your script, you should see no duplicates. Can you confirm that?
